### PR TITLE
[RFC] Don't truncate error messages

### LINKED ISSUE #5984: Messages from python calls get truncated
- NVIM 0.2.0-dev
- Ubuntu-Gnome 16.10
- gnome-terminal
- xterm-256color

### Actual behaviour

Messages from python calls get truncated. 

```
...
File "/usr/local/lib/python3.5/dist-packages/tasklib/backends.py", line 281, in execute_command                                                                         
    raise TaskWarriorException(error_msg)                                                                                                                                 
tasklib.backends.TaskWarriorException: Using alternate .taskrc file /home/someone/.taskrc                                                                                  
Configu
```
### Expected behaviour

Message continues after `Configu`

### PR DESCRIPTION
**Pull request by sander2**
This fixes #5984.

~~The functional test currently is not working; it freezes during execution.  I think this is because Neovim displays the `-- More -- SPACE/d/j: screen/page/line down, b/u/k: up, q: quit` line, which causes the `eval` not to be executed. I tried to do `feed('q')`, `feed('q<cr>')` and `feed('<cr>')`, but none of these help. How can I solve this?~~ The test is working properly now.

Currently, both the define `api_set_error` and the function `_api_set_error` are used from other files. Ideally, only the define would be used. However, not all code that writes to the error message actually sets the ErrorType, which is why the function is called instead. Should these occurrences be changed to actually set the correct type (in which case only the define is used from other files)?

Finally, a question about the performance. The `_api_set_error` always executes `vsnprintf` twice. Should I add a fast path for small error messages (e.g. allocate buffer on stack, print to that buffer, copy buffer to allocated space if the message actually fitted inside the temporary buffer)?

### CODE REVIEW

**Comment by justinmk** [comment]
> a question about the performance. The _api_set_error always executes vsnprintf twice. Should I add a fast path for small error messages (e.g. allocate buffer on stack, print to that buffer, copy buffer to allocated space if the message actually fitted inside the temporary buffer)?

That only happens if an error occurs, right? Which should almost never happen. So we don't need to optimize it, whatever makes the code clear is best.

File: src/nvim/msgpack_rpc/channel.c [hunk 1]
```diff
@@ -409,38 +408,38 @@ static void handle_request(Channel *channel, msgpack_object *request)
                channel->id);
       call_set_error(channel, buf);
     }
-    return;
-  }
```
Comment (justinmk) [inline src/nvim/msgpack_rpc/channel.c]: Why not keep the early return here? Would avoid moving all of the lines below.

**Comment by justinmk** [comment]
We probably _should_ truncate the error at _some_ limit, shouldn't we? Continued at  https://github.com/neovim/neovim/pull/6237

**Comment by Hjdskes** [comment]
@justinmk I think you mean #6559.

**Comment by justinmk** [comment]
Merged in #6559. Thanks a lot!

**Pull request closed without merging**